src/EnergyPlus/Pumps.cc

```diff
@@ -281,950 +274,907 @@ void GetPumpInput(EnergyPlusData &state)
-        if (!state.dataIPShortCut->lAlphaFieldBlanks(5) && !(state.dataPumps->PumpEquip(PumpNum).PumpScheduleIndex > 0)) {
-            ShowWarningError(state,
-                             std::string{RoutineName} + cCurrentModuleObject + "=\"" + state.dataPumps->PumpEquip(PumpNum).Name + "\", Invalid " +
-                                 state.dataIPShortCut->cAlphaFieldNames(5));
-            ShowContinueError(state, "Schedule named =[" + state.dataIPShortCut->cAlphaArgs(5) + "]. was not found and will not be used.");
-        }
-
-        state.dataPumps->PumpEquip(PumpNum).NomVolFlowRate = state.dataIPShortCut->rNumericArgs(1);
-        if (state.dataPumps->PumpEquip(PumpNum).NomVolFlowRate == AutoSize) {
-            state.dataPumps->PumpEquip(PumpNum).NomVolFlowRateWasAutoSized = true;
-        }
-        state.dataPumps->PumpEquip(PumpNum).NomPumpHead = state.dataIPShortCut->rNumericArgs(2);
-        state.dataPumps->PumpEquip(PumpNum).NomPowerUse = state.dataIPShortCut->rNumericArgs(3);
-        if (state.dataPumps->PumpEquip(PumpNum).NomPowerUse == AutoSize) {
-            state.dataPumps->PumpEquip(PumpNum).NomPowerUseWasAutoSized = true;
-        }
-        state.dataPumps->PumpEquip(PumpNum).MotorEffic = state.dataIPShortCut->rNumericArgs(4);
-        state.dataPumps->PumpEquip(PumpNum).FracMotorLossToFluid = state.dataIPShortCut->rNumericArgs(5);
-        state.dataPumps->PumpEquip(PumpNum).PartLoadCoef(1) = state.dataIPShortCut->rNumericArgs(6);
-        state.dataPumps->PumpEquip(PumpNum).PartLoadCoef(2) = state.dataIPShortCut->rNumericArgs(7);
-        state.dataPumps->PumpEquip(PumpNum).PartLoadCoef(3) = state.dataIPShortCut->rNumericArgs(8);
-        state.dataPumps->PumpEquip(PumpNum).PartLoadCoef(4) = state.dataIPShortCut->rNumericArgs(9);
-        state.dataPumps->PumpEquip(PumpNum).MinVolFlowRate = state.dataIPShortCut->rNumericArgs(10);
-        if (state.dataPumps->PumpEquip(PumpNum).MinVolFlowRate == AutoSize) {
-            state.dataPumps->PumpEquip(PumpNum).minVolFlowRateWasAutosized = true;
-        } else if (!state.dataPumps->PumpEquip(PumpNum).NomVolFlowRateWasAutoSized &&
-                   (state.dataPumps->PumpEquip(PumpNum).MinVolFlowRate > state.dataPumps->PumpEquip(PumpNum).NomVolFlowRate)) {
+        if (!thisInput->cAlphaArgs(5).empty()) { // Initialized to zero, don't get a schedule for an empty
+            thisPump.PumpScheduleIndex = GetScheduleIndex(state, thisInput->cAlphaArgs(5));
+            if (thisPump.PumpScheduleIndex <= 0) {
+                ShowWarningError(state,
+                                 format("{}{}=\"{}\", Invalid {}", RoutineName, cCurrentModuleObject, thisPump.Name, thisInput->cAlphaFieldNames(5)));
+                ShowContinueError(state, format("Schedule named =[{}]. was not found and will not be used.", thisInput->cAlphaArgs(5)));
+            }
+        }
+
+        thisPump.NomVolFlowRate = thisInput->rNumericArgs(1);
```

Comment:
Interesting, I hadn't thought of aliasing the DataIPShortcut.  I would suggest that in the future, instead of doing that, just remove the usage of DataIPShortcut.  The purpose of that module was to hold a set of global arrays that each developer could use without having to manage and resize arrays themselves.  In C++, we can just pass vectors in and out of the input processor routines without having to presize them.  Then you can just use local variables named rNumericArgs, etc.  And there won't be any need to alias anything.